contracts/interchain-token-service/tests/flow_limit.rs

```diff
@@ -0,0 +1,271 @@
+    types::{HubMessage, InterchainTransfer, Message},
+    InterchainTokenServiceClient,
+};
+use soroban_sdk::testutils::{Address as _, Ledger as _};
+use soroban_sdk::{vec, xdr::ToXdr, Address, Bytes, BytesN, Env, String, Vec};
+use utils::{
+    approve_gateway_messages, register_chains, setup_env, setup_gas_token, setup_its_token,
+    HUB_CHAIN,
+};
+
+const TEST_FLOW_LIMIT: Option<i128> = Some(1000);
+const EPOCH_TIME: u64 = 6 * 60 * 60;
+
+fn setup_flow_limit(env: &Env, client: &InterchainTokenServiceClient) -> (BytesN<32>, Address) {
+    let supply = i128::MAX;
+    let deployer = Address::generate(env);
+    let token_id = setup_its_token(env, client, &deployer, supply);
+
+    env.mock_all_auths();
+    client.set_flow_limit(&token_id, &TEST_FLOW_LIMIT);
+
+    (token_id, deployer)
+}
+
+fn create_interchain_transfer_message(
+    env: &Env,
+    client: &InterchainTokenServiceClient,
+    token_id: &BytesN<32>,
+    amount: i128,
+) -> (String, String, String, Bytes, Vec<GatewayMessage>) {
+    let sender = Address::generate(env).to_xdr(env);
+    let recipient = Address::generate(env).to_xdr(env);
+    let source_chain = client.its_hub_chain_name();
+    let source_address = Address::generate(env).to_string();
+
+    let msg = HubMessage::ReceiveFromHub {
+        source_chain: String::from_str(env, HUB_CHAIN),
+        message: Message::InterchainTransfer(InterchainTransfer {
+            token_id: token_id.clone(),
+            source_address: sender,
+            destination_address: recipient,
+            amount,
+            data: None,
+        }),
+    };
+    let payload = msg.abi_encode(env).unwrap();
+    let payload_hash: BytesN<32> = env.crypto().keccak256(&payload).into();
+
+    let message_id = Address::generate(env).to_string();
+
+    let messages = vec![
+        &env,
+        GatewayMessage {
+            source_chain: source_chain.clone(),
+            message_id: message_id.clone(),
+            source_address: source_address.clone(),
+            contract_address: client.address.clone(),
+            payload_hash,
+        },
+    ];
+
+    (source_chain, message_id, source_address, payload, messages)
+}
+
+#[test]
+fn set_flow_limit_succeeds() {
+    let (env, client, _, _, _) = setup_env();
+    let token_id = BytesN::from_array(&env, &[1; 32]);
+
+    assert_eq!(client.flow_limit(&token_id), None);
+
+    assert_invoke_auth_ok!(
+        client.operator(),
+        client.try_set_flow_limit(&token_id, &TEST_FLOW_LIMIT)
+    );
+
+    assert_eq!(client.flow_limit(&token_id), TEST_FLOW_LIMIT);
+
+    goldie::assert!(events::fmt_last_emitted_event::<FlowLimitSetEvent>(&env));
+}
+
+#[test]
+fn set_flow_limit_fails_invalid_amount() {
+    let (env, client, _, _, _) = setup_env();
+    let token_id = BytesN::from_array(&env, &[1; 32]);
+
+    let invalid_limit = Some(-1);
+
+    env.mock_all_auths();
+
+    assert_contract_err!(
+        client.try_set_flow_limit(&token_id, &invalid_limit),
```

Comment:
```suggestion
        client.mock_all_auths().try_set_flow_limit(&token_id, &invalid_limit),
```
Only mock the specific call